### PR TITLE
Voice Controlled Car

### PR DESCRIPTION
**Pull request by praveenkumar1410**
## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Review by bellesea — CHANGES_REQUESTED** [review]
hiii @praveenkumar1410 omg this is aewsome!! 

are you referencing any guides / existing voice controlled cars for this?

and, what's the role of the pi4 in this project?

also, what do you mean by "generate a g-code for the image and then run the machine"? and, is there an existing tool that allows you to parse the voice commands into instructions for the car or will you be coding that yourself?

thank you so much!!

**Comment by praveenkumar1410** [comment]
> hiii @praveenkumar1410 omg this is aewsome!!
> 
> are you referencing any guides / existing voice controlled cars for this?
> 
> and, what's the role of the pi4 in this project?
> 
> also, what do you mean by "generate a g-code for the image and then run the machine"? and, is there an existing tool that allows you to parse the voice commands into instructions for the car or will you be coding that yourself?
> 
> thank you so much!!

Hii ,I'm @praveenkumar1410 

Yes , I am refering an existing product. The project is available in arduino.

The pi4 helps to connect with wifi , so that it will be helpful for me to control it using mobile application( which as joy stick options)/ I can also control it using internet. 

The same car can be used for many innovative ideas with pi4.

"generate a g-code for the image and then run the machine" - I recently heard about it generating code when we specify actions. But I haven't used it . I am going to code on my own.

Thank you😊

**Review by praveenkumar1410 — COMMENTED** [review]
updated praveenkumar1410.md

**Comment by bellesea** [comment]
@praveenkumar1410 great! thanks for adding this, could you please link the existing project that you're modelling your car after within your file?

and, could you please update your plan to show when you'd use the Pi, how it integrates with your car, and steps you'll need to do to allow you to control it from your mobile app?

also, how will your dashboard generate g-code? 

and, i haven't looked that much into how voice controlled cars work... would you be willing to expand on how your car would follow g-code instructions? 

please let me know if you have any questions!

**Comment by praveenkumar1410** [comment]
Hii @bellasea Thank you for notifying. 

Sorry for inconvenience.

I have changed my document in response to your request.

There is no generation of G-code. I am coding on my own. I removed that part from my document. 

Please check it out.

File: praveenkumar1410.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+3.) Plug the monitor (via HDMI), mouse, keyboard, Arduino Uno (via USB), and WiFi adapter (if applicable) into the Raspberry Pi.
+
+4.) Power on the Raspberry Pi
+
+5.) When the Raspberry Pi finishes booting, it will ask you to connect to WiFi.
+
+## Existing Project
+
+https://create.arduino.cc/projecthub/mathwuy/voice-controlled-k-nex-car-f1b867?ref=similar&ref_id=443067&offset=4
```
Comment (bellesea) [inline praveenkumar1410.md:41]: for some reason whenever i go to this link it redirects to https://projecthub.arduino.cc/ :( would you be able to share a downloaded pdf or smth of the page so we could view it as reference?

File: praveenkumar1410.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+| L293D H-Bridge Dual Motor Driver IC   | https://amzn.to/2WmKsek                                                                        | $10.99 |
+| Robot Chasis & Wheels                 | https://amzn.to/3bq6tNo                                                                        | $19.99 |
+| Jumper Wires     (Q-20)               | https://amzn.to/2L8Xc1p                                                                        | $1.20  |
+| Breadboard                            | https://amzn.to/2YM6YyS                                                                        | $11.98 |
+| Raspberry Pi 4                        | https://bit.ly/3GaOxZY                                                                         | $47.15 |
+| Total                                 |                                                                                                | $200.98|
+
+No shipping costs are included because the prices included in the table is very similar to the prices in my local electronic store. I can confirm every item listed here is available in the store and don't need to ship them from overseas.
```
Comment (bellesea) [inline praveenkumar1410.md:58]: does the budget include taxes?

File: praveenkumar1410.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+10. Complete the project if everything work as expected.
+11. Connect the Raspberry pi4. The script on the Raspberry Pi that handles the processing of requests is written in Python.When the python script is activated, this opens the link between the Arduino Uno and the Raspberry Pi, thus enabling the car's motors and lights.This script also opens a communication link to a servo connected to pin 18 on the Raspberry Pi. This servo controls the turning mechanism in the car. The script uses PubNub to retrieve Alexa's requests and feeds these requests into a series of if statements. Depending on the command, the script will either send the command to the Arduino Uno through serial communication, tell the servo to turn the car, or return an unknown command error.
+
+## Raspberri Pi4 SET UP
+
+1.) Download the Raspberry Pi K'nex Car Control image.
+
+2.) flash the 16 gb sd card with the image.
+
+3.) Plug the monitor (via HDMI), mouse, keyboard, Arduino Uno (via USB), and WiFi adapter (if applicable) into the Raspberry Pi.
```
Comment (bellesea) [inline praveenkumar1410.md:33]: do you already have all of these?

File: praveenkumar1410.md (hunk 1 above)
Comment (bellesea) [reply praveenkumar1410.md:41]: is it the same as https://www.hackster.io/mathwuy/voice-controlled-k-nex-car-f1b867?

**Review by bellesea — CHANGES_REQUESTED** [review]
@praveenkumar14 apologies for the delayed response! thanks sm for pushing these new changes :))) just a few more questions hehe :)

thanks!

**Comment by cfanoulis** [comment]
Hi @praveenkumar1410! Thanks for your proposal. It's been a while since we've had any news from you and we want to know if you're still interested in the grant. Please reply in the next 48 hours to this PR (or this email) if you're still interested, else we'll be closing this proposal. Hacky New Year!

**Comment by cfanoulis** [comment]
Hello! Unfortunately 2 days have passed, which means that I'll be closing this PR as stale. If you still wish to pursue a grant please let us know by replying to this PR before Jan. 15th

**Pull request closed without merging**